sdk/src/types/block/core/mod.rs

```diff
@@ -81,7 +82,10 @@ impl BlockBody {
 
     /// Creates a new [`BasicBlockBodyBuilder`].
     #[inline(always)]
-    pub fn build_basic(strong_parents: self::basic::StrongParents, max_burned_mana: u64) -> BasicBlockBodyBuilder {
+    pub fn build_basic(
```

Comment:
Maybe I'm annoying now but I would do the same as the output and provide two builders. Having a `MaxBurnedManaAmount` as param is not a very great UX IMO.

Reply:
You're right. Changed to something more convenient and less verbose.

Reply:
That's better than what I had in mind, maybe we could do the same for outputs

Reply:
(another PR if ever)

Reply:
Even though it's still nice to be explicit on the name

Reply:
yeah ... only problem with that is you'ld have to adapt (possible add) builders when the enum changes.